File: src/main/java/no/fintlabs/kafka/topic/name/TopicNamePatternRegexUtils.java
package no.fintlabs.kafka.topic.name;

import java.util.Collection;
import java.util.StringJoiner;

class TopicNamePatternRegexUtils {

    static StringJoiner createPartialTopicPatternJoiner() {
        return new StringJoiner("\\.");
    }

    static StringJoiner createTopicPatternJoiner() {
        return new StringJoiner("\\.", "^", "$");
    }

    static String any() {
        return "[^.]+";
    }

    static String anyOf(Collection<String> values) {
        return "(" + String.join("|", values) + ")";
    }

    static String startingWith(String value) {
        return value + "[^.]*";
    }

    static String endingWith(String value) {
        return "[^.]*" + value;
    }

    static String containing(String value) {
        return "[^.]*" + value + "[^.]*";
    }

}

File: src/main/kotlin/no/novari/kafka/topic/name/TopicNamePatternRegexUtils.kt
package no.novari.kafka.topic.name

import java.util.StringJoiner

object TopicNamePatternRegexUtils {
    @JvmStatic
    fun createTopicPatternJoiner(): StringJoiner = StringJoiner("\\.", "^", "$")

    @JvmStatic
    fun any(): String = "[^.]+"

    @JvmStatic
    fun anyOf(values: Collection<String>): String = "(${values.joinToString("|")})"

    @JvmStatic
    fun startingWith(value: String): String = "$value[^.]*"

    @JvmStatic
    fun endingWith(value: String): String = "[^.]*$value"

    @JvmStatic
    fun containing(value: String): String = "[^.]*$value[^.]*"
}
